aws/request_id.go

```diff
@@ -0,0 +1,28 @@
+package aws
+
+import (
+	"github.com/awslabs/smithy-go/middleware"
+)
+
+// requestID is used to retrieve request id from response metadata
+type requestID struct {
+}
+
+// SetRequestIDMetadata sets the provided request id over middleware metadata
+func SetRequestIDMetadata(metadata *middleware.Metadata, id string) {
+	metadata.Set(requestID{}, id)
+}
+
+// GetRequestIDMetadata retrieves the request id from middleware metadata
+func GetRequestIDMetadata(metadata middleware.Metadata) string {
```

Comment:
Did we already discuss if these getters should return `(T, bool)` to indicated empty value vs unset?